src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp

```diff
@@ -484,107 +485,153 @@ void HydrofractureSolver::SetupDofs( DomainPartition const & domain,
                                        DofManager & dofManager,
-                                       CRSMatrix< real64, globalIndex > & GEOSX_UNUSED_PARAM( localMatrix ),
-                                       array1d< real64 > & GEOSX_UNUSED_PARAM( localRhs ),
-                                       array1d< real64 > & GEOSX_UNUSED_PARAM( localSolution ),
+                                       CRSMatrix< real64, globalIndex > & localMatrix,
+                                       array1d< real64 > & localRhs,
+                                       array1d< real64 > & localSolution,
                                        bool const setSparsity )
 {
   GEOSX_MARK_FUNCTION;
 
+  GEOSX_UNUSED_VAR( setSparsity );
+
   MeshLevel & mesh = *domain.getMeshBody( 0 )->getMeshLevel( 0 );
   m_flowSolver->ResetViews( mesh );
 
-  m_solidSolver->SetupSystem( domain,
-                              m_solidSolver->getDofManager(),
-                              m_solidSolver->getLocalMatrix(),
-                              m_solidSolver->getLocalRhs(),
-                              m_solidSolver->getLocalSolution() );
-
-  m_flowSolver->SetupSystem( domain,
-                             m_flowSolver->getDofManager(),
-                             m_flowSolver->getLocalMatrix(),
-                             m_flowSolver->getLocalRhs(),
-                             m_flowSolver->getLocalSolution(),
-                             setSparsity );
-
-  // setup coupled DofManager
-  m_dofManager.setMesh( domain, 0, 0 );
+  dofManager.setMesh( domain, 0, 0 );
+
   SetupDofs( domain, dofManager );
+  dofManager.reorderByRank();
 
-  // By not calling dofManager.reorderByRank(), we keep separate dof numbering for each field,
-  // which allows constructing separate sparsity patterns for off-diagonal blocks of the matrix.
-  // Once the solver moves to monolithic matrix, we can remove this method and just use SolverBase::SetupSystem.
-  m_matrix01.createWithLocalSize( m_solidSolver->getLocalMatrix().numRows(),
-                                  m_flowSolver->getLocalMatrix().numRows(),
-                                  9,
-                                  MPI_COMM_GEOSX );
-  m_matrix10.createWithLocalSize( m_flowSolver->getLocalMatrix().numRows(),
-                                  m_solidSolver->getLocalMatrix().numRows(),
-                                  24,
-                                  MPI_COMM_GEOSX );
-
-#if 0
-  dofManager.setSparsityPattern( m_matrix01, keys::TotalDisplacement, FlowSolverBase::viewKeyStruct::pressureString );
-  dofManager.setSparsityPattern( m_matrix10, FlowSolverBase::viewKeyStruct::pressureString, keys::TotalDisplacement );
-#else
+  localIndex const numLocalRows = dofManager.numLocalDofs();
 
-  NodeManager const & nodeManager = *mesh.getNodeManager();
-  ElementRegionManager const & elemManager = *mesh.getElemManager();
+  SparsityPattern< globalIndex > patternOriginal;
+  dofManager.setSparsityPattern( patternOriginal );
```

Comment:
what portion is being set here? `disp-disp` and  `pressure-pressure` ?

Reply:
`disp-disp`, `pressure-pressure`, `pressure-disp`,  and element-wise `pressure-disp`. I had to add manually the coupling `pressure-disp` withthe displacements of the neighbouring cells due to the aperture dependency of the transmissiblity.

Reply:
note that ultimately, I would like to general looping strategies for the sparsity and assembly just like is done for the FEM solver. That way we don't have to rely on a loop structure hidden inside the dof manager, and we can be assured that the loop used to fill sparsity is the same as for assembly.

Reply:
That would probably make it a lot clearer because as of now some parts of the sparsity are taken care by the dofManager but some are not and it is not that obvious. Although I should have some comments in there.